### ISSUE TITLE
OIDC username prefix option is not working

### ISSUE THREAD

**Issue by dimm0** [kubernetes/kubernetes#56169]
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
--oidc-username-prefix option for api-server seems to not have any effect.
The apiserver is running with --oidc-username-prefix=-, and if I'm changing ClusterRoleBinding username from prefix#username to username, I'm still getting:

```
$ kubectl get nodes
Error from server (Forbidden): nodes is forbidden: User "https://test.cilogon.org#http://cilogon.org/<user_id>" cannot list nodes at the cluster scope
```

**What you expected to happen**:
When using the "--oidc-username-prefix=-", the username should be "http://cilogon.org/<user_id>", without issuer prefix

**How to reproduce it (as minimally and precisely as possible)**:
Follow the steps to set up the OIDC authentication, add "--oidc-username-prefix=-" to apiserver config file, restart kubelet, create client config for OIDC, try "kubeadm get nodes"

**Anything else we need to know?**:
I'm wondering if users are cached somewhere?

**Environment**:
- Kubernetes version (use `kubectl version`):

Client Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.4", GitCommit:"9befc2b8928a9426501d3bf62f72849d5cbcd5a3", GitTreeState:"clean", BuildDate:"2017-11-20T19:11:02Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"8", GitVersion:"v1.8.4", GitCommit:"9befc2b8928a9426501d3bf62f72849d5cbcd5a3", GitTreeState:"clean", BuildDate:"2017-11-20T05:17:43Z", GoVersion:"go1.8.3", Compiler:"gc", Platform:"linux/amd64"}

- Cloud provider or hardware configuration:
kubeadm

- OS (e.g. from /etc/os-release):
centos 7.4

**Comment by ericchiang** [comment]
Re-opening for cherry pick.

**Comment by Josh-West-Bose** [comment]
The same issue is occurring with the `--oidc-groups-prefix` flag.

**Comment by ericchiang** [comment]
Fix has been cherry picked. Fix will be available in v1.8.5

/close
